File: Easy/1929_Concatenation_of_Array/Solution.java


public class Solution {
    public int[] getConcatenation(int[] nums) {
        int n = nums.length;
        int[] ans = new int[2 * n];

        for (int i = 0; i < n; i++) {
            ans[i] = nums[i];
            ans[i + n] = nums[i];
        }
        
        return ans;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] nums1 = {1, 2, 1};
        int[] result1 = solution.getConcatenation(nums1);
        for (int num : result1) {
            System.out.print(num + " ");
        }
        System.out.println();
        
        // test cases 2
        int[] nums2 = {1, 3, 2, 1};
        int[] result2 = solution.getConcatenation(nums2);
        for (int num : result2) {
            System.out.print(num + " ");
        }
        System.out.println();
        
    }
}

File: Easy/1929_Concatenation_of_Array/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(2 * n);

        for (int i = 0; i < n; i++) {
            ans[i] = ans[i + n] = nums[i];
        }

        return ans;
        
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    vector<int> nums1 = {1, 2, 1};
    vector<int> result1 = solution.getConcatenation(nums1);
    printVec(result1);
    
    // test cases 2
    vector<int> nums2 = {1, 3, 2, 1};
    vector<int> result2 = solution.getConcatenation(nums2);
    printVec(result2);

    return 0;
}